wordpress-mobile/WordPress-iOS
# Move the post lookup method out of PostService

## Changes

Similar to [the account lookup](https://github.com/wordpress-mobile/WordPress-iOS/pull/16168) and [the blog lookup](https://github.com/wordpress-mobile/WordPress-iOS/pull/15887) refactor, this PR moves the function that finds a post in a blog out of `PostService`.

## Test instructions

I don't see much risk in these changes. The lookup function implementation is almost a direct translation of the original Objective-C code, and the context objects used in call sites are not changed.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
One simple unit test.

PR submission checklist:
